components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/HTTPGetFunctionImpl.java

```diff
@@ -39,10 +39,21 @@ public HTTPGetFunctionImpl() {
     }
 
     @Override
-    public void httpGet(String epUrl, Map<String, Object> eventHandlers) {
+    public void httpGet(String epUrl, Map<String, Object> eventHandlers, Map<String, String>... optional) {
+    
+        // Take first optional parameter as headers else set header to null
+        Map<String, String> headers = optional.length >= 1 ? optional[0] : null;
```

Comment:
Yes, this approach is not the best way to solve this issue. I added a new PR introducing new methods `httpGetWithHeaders` and `httpPostWithHeaders` to accept headers as a required parameter instead of providing headers as an optional parameter to the existing HTTP methods.
[Pull request](https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/102)